app/src/main/java/com/infomaniak/mail/data/models/Folder.kt

```diff
@@ -137,20 +153,26 @@ class Folder : RealmObject {
 
     override fun hashCode(): Int = id.hashCode()
 
+    /**
+     * We've had a reference issue when modifying the `shouldDisplayDivider` of a Folder before sending it to the Adapter.
+     * Cloning it to do a deep copy resolved the issue.
+     */
+    public override fun clone() = super.clone() as Folder
```

Comment:
Couldn't do without it, but we moved it to a coroutine in the VM, instead of Main thread in the Adapter.

Reply:
A different coroutine is always welcome but this won't help with the memory consumption issue ... are we really not just doing something bad somewhere above where we get the folders ...? This is really unfortunate if we really can't do anything about it

Reply:
We looked with @tevincent and @sirambd but couldn't find a better way.

Reply:
To test: directly add the divider in the list, as an item.

Reply:
The clone has been removed, but the complexity forced me to do in in another PR : #1947.

Reply:
I feel like moving fixes and change requests across PRs doesn't always help and sometimes makes it harder to follow, but at least the issue is fixed and it's cool you could find an alternative to this `clone()` method